### PR TITLE
[controller][admin-tool] Add support for get-dead-stores API and admin tool command

### PR DESCRIPTION
**Pull request by MikeDafi**
## Problem Statement
 There was no public way for users to programmatically query which stores are considered "dead" by Venice.
     This made it difficult to surface lifecycle decisions driven by ACL access logs and discovery mechanisms.
     Additionally, there was no CLI toolchain integration for easily invoking or testing this logic in staging or production.

 ## Solution
 Introduced a new `GET_DEAD_STORES` endpoint in `ControllerRoute`, mapped it to the existing ACL-based evaluation logic.
 Added a method `getDeadStores` to `ControllerClient` which wraps the request to the controller.
 Implemented a matching CLI command in `AdminTool` that allows users to call this endpoint from the terminal using:
     ```
     ./venice-admin-tool.sh get-dead-stores -url <url> -cluster <cluster> [-store <store>] [-include_system_stores true]
     ```
     This enables debugging store lifecycle states, validating access audits, and surfacing potential deletions.

 ### Code changes
 - [x] Added `GET_DEAD_STORES` route to `ControllerRoute`
     - [x] Exposed `getDeadStores(...)` method in `ControllerClient`
     - [x] Added new admin tool command handler `GET_DEAD_STORES` with param parsing
 - [x] Printed `MultiStoreInfoResponse` object as output for clarity

 ### Logging
 - [ ] No new logs introduced directly in this diff. (Relies on controller logic already logging ACL decisions if needed)

 ## Does this PR introduce any user-facing or breaking changes?
     - [x] No breaking changes. This is an additive feature.
 - [x] Adds new CLI command and controller endpoint for observability and debugging

### CODE REVIEW

File: internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java [hunk 1]
```diff
@@ -1227,6 +1227,12 @@ public RepushJobResponse repushStore(String storeName) {
     return request(ControllerRoute.REPUSH_STORE, params, RepushJobResponse.class);
   }
 
+  public MultiStoreInfoResponse getDeadStores(String clusterName, String storeName, boolean includeSystemStores) {
+    QueryParams params =
+        newParams().add(CLUSTER, clusterName).add(NAME, storeName).add(INCLUDE_SYSTEM_STORES, includeSystemStores);
```
Comment (huangminchn) [inline internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java:1232]: Just noticed that storeName can be null since it's optional, it would be great to add a test case to ensure null could be passed all the way to backend.

Comment (MikeDafi) [reply internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java]: Oh you're right, now no longer passing the argument if it is indeed null

**Review by huangminchn — APPROVED** [review]
Thanks Michael!
